backend/api/parse_pdf.py

```
@@ -630,7 +865,52 @@ async def parse_pdf_with_progress_route(
+        progress_manager.progress_data[session_id] = session_data
+        debug_log(f"[PROGRESS] Session added to memory: {session_id}")
+    
+    # Final verification that session is available
+    debug_log(f"[PROGRESS] Final check - session in memory: {session_id in progress_manager.progress_data}")
+    if progress_manager.use_redis:
+        stored_data = progress_manager._get_progress_data(session_id)
+        debug_log(f"[PROGRESS] Final check - session in Redis: {stored_data is not None}")
+    
+    # Start the actual parsing in the background
+    import asyncio
+    asyncio.create_task(parse_pdf_with_progress(file, context_files, save_to_db, session_id, db))
```

Ruff (0.13.3)

Store a reference to the return value of `asyncio.create_task`

(RUF006)